### ISSUE TITLE
[Helm/Release] values are not compatible with helm chart values

### ISSUE THREAD

**Issue by gitfool** [pulumi/pulumi-kubernetes#1726]
Using the dotnet/c# sdk, I tried switching from helm charts to helm releases but the values are not compatible since it doesn't support serializing anonymous types, so although it compiles, it fails at runtime.

## Steps to reproduce

1. Run [gitfool/Pulumi.Dungeon/test](https://github.com/gitfool/Pulumi.Dungeon/commits/test) using helm charts @ [e0354be](https://github.com/gitfool/Pulumi.Dungeon/commit/e0354bee7ac9f0d75a14338450c018e501a7488c) which succeeds ([test-k8s-charts.log](https://github.com/pulumi/pulumi-kubernetes/files/7190948/test-k8s-charts.log))
2. Run [gitfool/Pulumi.Dungeon/test](https://github.com/gitfool/Pulumi.Dungeon/commits/test) using helm releases @ [a1782b5](https://github.com/gitfool/Pulumi.Dungeon/commit/a1782b546f272b05554e852db551789e1a08c0bf) which fails ([test-k8s-releases.log](https://github.com/pulumi/pulumi-kubernetes/files/7190950/test-k8s-releases.log))

This is just the first failure of many due to the use of anonymous types, for example with [cert-manager](https://github.com/gitfool/Pulumi.Dungeon/blob/a1782b546f272b05554e852db551789e1a08c0bf/K8s/K8sStack.cs#L216-L220):
```c#
var certManagerValues =
    new Dictionary<string, object>
    {
        ["prometheus"] = new
        {
            enabled = true,
            servicemonitor = new { enabled = true }
        }
    };
```
This would work for helm releases if I refactored all nested values to use dictionaries:
```c#
var certManagerValues =
    new Dictionary<string, object>
    {
        ["prometheus"] = new Dictionary<string, object>
        {
            ["enabled"] = true,
            ["servicemonitor"] = new Dictionary<string, object> { ["enabled"] = true }
        }
    };
```
But that is not desirable and I expected helm releases to be a drop in replacement for helm charts, from an api pov, and being able to use anonymous types in c# is essential to reduce the ceremony required to specify helm values. (I've included lots of common helm charts for reference.)

**Comment by mikhailshilkov** [comment]
Thank you for reporting this @gitfool - that's a legit concern. I suspect this distinction comes from the difference in the implementation where the `Chart` component is implemented purely in .NET and is able to use `Values` in-process, while `Release` is a multi-language resource, so the `Values` are serialized across the process boundary and that's where anonymous objects fail.

We would need to change our core .NET SDK to support that. @t0yv0 does that makes sense to you? How hard would it be to amend serialization to support anonymous objects?

**Comment by t0yv0** [comment]
>  @t0yv0 does that makes sense to you? How hard would it be to amend serialization to support anonymous objects?

Yeah the problem statement makes sense. I've only glanced at the current serde code in .NET SDK, still need to dive deeper, but it should be quite doable.
